parastell/nwl_utils.py

```diff
@@ -215,192 +218,213 @@ def extract_coords(source_file):
     return coords
 
 
-def plot(nwl_mat, phi_pts, theta_pts, num_levels):
-    """Generates contour plot of NWL.
-
-    Arguments:
-        nwl_mat (array of array of float): NWL solutions at centroids of
-            (phi, theta) bins (MW).
-        phi_pts (array of float): centroids of toroidal angle bins (rad).
-        theta_bins (array of float): centroids of poloidal angle bins (rad).
-        num_levels (int): number of contour regions.
-    """
-    phi_pts = np.rad2deg(phi_pts)
-    theta_pts = np.rad2deg(theta_pts)
-
-    levels = np.linspace(np.min(nwl_mat), np.max(nwl_mat), num=num_levels)
-    fig, ax = plt.subplots()
-    CF = ax.contourf(phi_pts, theta_pts, nwl_mat.T, levels=levels)
-    cbar = plt.colorbar(CF)
-    cbar.ax.set_ylabel("NWL (MW/m2)")
-    plt.xlabel("Toroidal Angle (degrees)")
-    plt.ylabel("Poloidal Angle (degrees)")
-    fig.savefig("NWL.png")
-
-
-def area_from_corners(corners):
-    """
-    Calculate an approximation of the area defined by 4 xyz points
+def compute_quadrilateral_area(corners):
+    """Approximates the area of a non-planar region bounded by four points.
 
     Arguments:
-        corners (4x3 numpy array): list of 4 (x,y,z) points. Connecting the
-            points in the order given should result in a polygon
+        corners (numpy.array): Cartesian coordinates of four points. Connecting
+            the points in the order given should result in a polygon.
 
     Returns:
-        area (float): approximation of area
+        area (float): approximation of area.
     """
-    # triangle 1
-    v1 = corners[3] - corners[0]
-    v2 = corners[2] - corners[0]
-
-    v3 = np.cross(v1, v2)
+    # Triangle 1
+    edge_1 = corners[3] - corners[0]
+    edge_2 = corners[2] - corners[0]
+    normal_vector = np.cross(edge_1, edge_2)
+    area_1 = np.linalg.norm(normal_vector) / 2
 
-    area1 = np.sqrt(np.sum(np.square(v3))) / 2
+    # Triangle 2
+    edge_1 = corners[1] - corners[0]
+    edge_2 = corners[2] - corners[0]
+    normal_vector = np.cross(edge_1, edge_2)
+    area_2 = np.linalg.norm(normal_vector) / 2
 
-    # triangle 2
-    v1 = corners[1] - corners[0]
-    v2 = corners[2] - corners[0]
+    total_area = area_1 + area_2
 
-    v3 = np.cross(v1, v2)
+    return total_area
 
-    area2 = np.sqrt(np.sum(np.square(v3))) / 2
 
-    area = area1 + area2
-
-    return area
-
-
-def nwl_plot(
+def compute_nwl(
     source_file,
-    ss_file,
-    plas_eq,
-    tor_ext,
-    pol_ext,
+    vmec_file,
     wall_s,
-    num_phi=101,
-    num_theta=101,
-    num_levels=10,
+    toroidal_extent,
+    neutron_power,
+    num_toroidal_bins=101,
+    num_poloidal_bins=101,
+    conv_tol=1e-6,
+    num_batches=1,
     num_crossings=None,
-    chunk_size=None,
-    num_threads=1,
+    num_threads=None,
+    logger=None,
 ):
-    """Computes and plots NWL. Assumes toroidal extent is less than 360 degrees
+    """Computes and plots NWL. Assumes toroidal extent is less than 360 degrees.
```

Comment:
Plotting now happens elsewhere
```suggestion
    """Computes NWL. Assumes toroidal extent is less than 360 degrees.
```